src/core/math/blue-noise.js

```diff
@@ -0,0 +1,50 @@
+
+let data = null;
+const initData = () => {
+    if (!data) {
+        const binaryString = atob(base64String);
+
+        // the data stores an array of RGBA 8-bit values, each of the color channels represents
+        // a separate blue noise stream
+        data = Uint8Array.from(binaryString, char => char.charCodeAt(0));
+    }
+};
+
+const blueNoiseData = () => {
+    initData();
+    return data;
+};
+
+/**
+ * Blue noise based random numbers API.
+ *
+ * @ignore
+ */
+class BlueNoise {
+    seed = 0;
+
+    constructor(seed = 0) {
+        this.seed = seed * 4;
+        initData();
+    }
+
+    next() {
+        this.seed = (this.seed + 4) % data.length;
+    }
+
+    value() {
+        this.next();
+        return data[this.seed] / 255;
+    }
+
+    vec4(dest = new Vec4()) {
+        this.next();
+        return dest.set(data[this.seed], data[this.seed + 1], data[this.seed + 2], data[this.seed + 3]).mulScalar(1 / 255);
```

Comment:
This will error if `this.seed` is currently `data.length - 1 or 2 or 3`. Either next must modulo `data.length - 4` or these accesses require %.

Reply:
this.seed is always pointing to a start value of the 4 (RGBA), so there are always 3 following values available